crates/forge/bin/cmd/script/build.rs

```diff
@@ -346,5 +280,5 @@ pub struct BuildOutput {
     pub highlevel_known_contracts: ArtifactContracts<ContractBytecodeSome>,
     pub libraries: Libraries,
     pub predeploy_libraries: Vec<ethers::types::Bytes>,
-    pub sources: BTreeMap<u32, String>,
+    pub sources: HashMap<String, HashMap<u32, (String, ContractBytecodeSome)>>,
```

Comment:
Almost sure this might cause linking problems, ordering might be important here

Reply:
Could you please elaborate ? I'm not sure why ordering would matter because we map the sources from the artifact Id https://github.com/foundry-rs/foundry/pull/5753/files/466d24a4be210ff3be7b7eedb8e62c76e4630d6e#diff-fdebd782f78bd93139131fe4f614d8ab47564bceef84dd9a009483b0c5ac7b41R42